src/functions/findFirst.js

```diff
@@ -149,20 +150,19 @@ export const findFirst = wrap({
             hi.assert(result.high === 4);
         },
         "emptySearchString": (hi) => {
+            hi.assertUndefined(hi("test").findFirst(""));
+        },
+        "unboundedSearchInput": (hi) => {
             hi.assertFailWith(NotBoundedError, () => {
-                return hi("test").findFirst("");
+                return hi("test").findFirst(hi.repeatElement({
+                    key: "hello",
+                    value: "world"
+                }));
             });
         },
         "nonStringSearchSequence": (hi) => {
```

Comment:
This function is valid for substrings other than honest-to-god string literals. Rename test case to `emptySearchSequence` perhaps? Add also a blank string input to go with the others?